libs/colbert/ragstack_colbert/colbert_embedding_model.py

```diff
@@ -53,165 +25,87 @@ class ColbertEmbeddingModel(BaseEmbeddingModel):
-            Distributed(self.__nranks)
-            logging.info("distribution initialization completed")
+        if query_maxlen is None:
+            query_maxlen = -1
 
-        with Run().context(RunConfig(nranks=nranks)):
-            if self.__cuda:
-                torch.cuda.empty_cache()
-            self.colbert_config = ColBERTConfig(
-                doc_maxlen=doc_maxlen,
-                nbits=nbits,
-                kmeans_niters=kmeans_niters,
-                nranks=self.__nranks,
-                checkpoint=checkpoint,
-                query_maxlen=query_maxlen,
-                gpus=total_visible_gpus,
-            )
-        logging.info("creating checkpoint")
-        self.checkpoint = Checkpoint(
-            self.colbert_config.checkpoint,
-            colbert_config=self.colbert_config,
-            verbose=verbose,
-        )
-        self.encoder = CollectionEncoder(
-            config=self.colbert_config, checkpoint=self.checkpoint
+        self._query_maxlen = query_maxlen
+        self._chunk_batch_size = chunk_batch_size
+
+        colbert_config = ColBERTConfig(
+            doc_maxlen=doc_maxlen,
+            nbits=nbits,
+            kmeans_niters=kmeans_niters,
+            nranks=nranks,
+            checkpoint=checkpoint,
         )
-        self.query_tokenizer = QueryTokenizer(self.colbert_config)
-        if self.__cuda:
-            self.checkpoint = self.checkpoint.cuda()
+        self._encoder = TextEncoder(config=colbert_config, verbose=verbose)
 
-    def embed_chunks(
-        self, chunks: List[ChunkData], doc_id: Optional[str] = None
-    ) -> List[EmbeddedChunk]:
+    # implements the Abstract Class Method
+    def embed_texts(self, texts: List[str]) -> List[Embedding]:
         """
-        Encodes a list of text chunks into embeddings, returning them as a list of EmbeddedChunk objects.
-        Each chunk text is converted into a dense vector representation.
+        Embeds a list of texts into their corresponding vector embedding representations.
 
         Parameters:
-            texts (List[str]): The list of chunk texts to be embedded.
-            doc_id (Optional[str]): An optional document identifier. If not provided, a UUID is generated.
+            texts (List[str]): A list of string texts.
 
         Returns:
-            List[EmbeddedChunk]: A list of EmbeddedChunk objects containing the embeddings and document/chunk identifiers.
+            List[Embedding]: A list of embeddings, in the order of the input list
         """
 
-        if doc_id is None:
-            doc_id = str(uuid.uuid4())
-
-        timeout = 60 + len(chunks)
-
-        chunk_data_map: Dict[int, ChunkData] = {}
-        texts: List[str] = []
-
-        for chunk_idx, chunk in enumerate(chunks):
-            chunk_data_map[chunk_idx] = chunk
-            texts.append(chunk.text)
+        chunks = [
+            Chunk(doc_id="dummy", chunk_id=i, text=t) for i, t in enumerate(texts)
```

Comment:
Shouldn't we generate uuid as the doc_id? This was implemented.

Reply:
we still do, just not here.  See: https://github.com/datastax/ragstack-ai/pull/407/files#diff-f5761527a38ba7b646140fe7d5b2ddca8c5a508bcb9da6c7670144a18c917f40R88-R89

this class no longer has any knowledge of doc_id... here it is just using the shared Chunk object for chunk_id management.